Overhaul to kriging methods, data structure, and primary `Survey`-class methods (#241)
OSOceanAcoustics/echopop: docs/core_data_structure.md

(data-structure)=
# Core data structures

> Input data, intermediate variables and calculations, and results are stored within several attributes associated with the `Survey`-class object. Information on how these attributes are used in the general `echopop` workflow can be found in greater detail [within the example notebook](echopop-example-workflow). 

Reviewer: This is ok for now, but I think we should have a diagram or table on the top of this page as a quick reference. I'll add an issue to track this.

Reviewer: Oh I had missed what you have as a TOC since it didn't get rendered in GH, but it got rendered in RTD -- awesome! I would just suggest some edits here then!
```suggestion
Input data, intermediate variables and calculations, and results are stored within several attributes associated with the `Survey`-class object. The attributes and stored data are detailed in the sections below. See [the example notebook](echopop-example-workflow) for information on how these attributes are used in the general `echopop` workflow. 
```

## Class attributes

There are currently **five** unique class attributes that store different types of information and data. 

* `Survey.meta`: Necessary information such as the date the object was created and general data workflow/provenance would be collected.
* `Survey.config`: Background configuration settings. 
* `Survey.input`: Imported acoustic, biological, kriging, and stratification [data detailed elsewhere](input-files).
* `Survey.analysis`: Intermediate data products, variables, and other calculations that may be both relevant to users (e.g. number proportions) or are required for computing the final results. 
* `Survey.results`: Overall results for each analysis.

> Each of these attributes are organized using nested dictionaries that may contain an uneven number of levels. For instance, calculated number and weight proportions computed via the transect analysis are nested via the following structure:

Reviewer: ```suggestion
Each of these attributes are organized using nested dictionaries that may contain different number of levels. For instance, number and weight proportions computed via the `transect_analysis` method are nested in the following structure:
```

* Survey object
  * `analysis` attribute
    * `transect` key
      * `biology` sub-key
        * `proportions` sub-key
          * `number` sub-key
            * Various number proportion `DataFrame`'s
          * `weight` sub-key

>             * Various weight proportion `DataFrame`'s

Reviewer: I would suggest removing the attribute and sub-key specifiers to avoid confusion. These are all dictionary keys all the down to each of the "leaves", and as you know, key-value has very specific meaning in python.

> which is equivalent to the following Python code:

Reviewer: ```suggestion
which can be accessed using the following Python code:
```

#### Acoustic data (`['acoustics']`)
- `nasc_df`: Georeferenced acoustic trawl data for all-aged and age-2+ fish.

#### Biological data (`['biology']`)
- `catch_df`: Total unaged weights for each haul.
- `distributions`: Age and length distributions.
  - `age_bins_df`: Age distribution/histogram bins.
  - `length_bins_df`: Length distribution/histogram bins.
- `haul_to_transect_df`: Haul-to-transect key that links haul numbers to their respective transects.
- `length_df`: Unaged (Station 1) length measurements.
- `specimen_df`: Aged (Station 2) length and weight measurements.

#### Spatial data (`'[spatial]'`)
- `strata_df`: The length-based 'KS' stratum definitions and fraction of hake for each haul.
- `geo_strata_df`: The latitudinal extents/ranges of each KS stratum throughout the survey region.
- `inpfc_strata_df`: The INPFC stratum definitions and their respective latitudinal limits

> #### Statistical data (`'[statistics]'`)

Reviewer: ```suggestion
#### Statistical data (`['statistics']`)
```
